### PR TITLE
Pr/args fixes

### PR DESCRIPTION
**Pull request by uli42**
fixes and cleanups for args.c

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/Args.c [hunk 1]
```diff
@@ -1394,30 +1394,6 @@ static void nxagentParseSingleOption(char *name, char *value)
...
-    else if ((strcmp(value, "none") == 0) || (strcmp(value, "0") == 0))
-    {
-      nxagentChangeOption(Clipboard, ClipboardNone);
-    }
-    else
-    {
-      nxagentChangeOption(Clipboard, ClipboardBoth);
-    }
-    return;
-  }
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Args.c:1419]: Removing this is not a good idea, I guess. It removes functionality for setting clipboard security via nx/nx options. Please fix this.

**Review by sunweaver — CHANGES_REQUESTED** [review]
See inline comment about nx/nx option "clipboard" code block removal. Please comment or fix.

**Comment by uli42** [comment]
No, it does not, because the identical code is found elsewhere, too. This
is here is never reached

Mike Gabriel <notifications@github.com> schrieb am Fr., 1. Nov. 2019, 16:29:

File: nx-X11/programs/Xserver/hw/nxagent/Args.c (hunk 1 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Args.c:1419]: No, it does not, because the identical code is found elsewhere, too. This here is never reached

**Comment by sunweaver** [comment]
Hi Uli,

On  Fr 01 Nov 2019 17:43:09 CET, Ulrich Sibiller wrote:

> uli42 commented on this pull request.
>
>
>
>> -      nxagentChangeOption(Clipboard, ClipboardClient);
> -    }
> -    else if (strcmp(value, "server") == 0)
> -    {
> -      nxagentChangeOption(Clipboard, ClipboardServer);
> -    }
> -    else if ((strcmp(value, "none") == 0) || (strcmp(value, "0") == 0))
> -    {
> -      nxagentChangeOption(Clipboard, ClipboardNone);
> -    }
> -    else
> -    {
> -      nxagentChangeOption(Clipboard, ClipboardBoth);
> -    }
> -    return;
> -  }
>
> No, it does not, because the identical code is found elsewhere, too.  
> This here is never reached

that code you may reference is for command line args parsing.

Please give me line numbers where you think this block is and where it  
should be removed (using current 3.6.x branch).

I still think, you are mixing up nx/nx option parsing and command line  
args parsing.

Thanks,
Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by uli42** [comment]
The checks for command line options and nx/nx option are mixed. 

nxagentParseOptionString will call nxagentParseSingleOption for every option found in the options string which will call ddxProcessArguments with a "-" prepended for these options (meaning they are also allowed on the command line; for options that are only allowed in the nx/nx options nxagentParseSingleOption will process the option and return): 

```
   1042   if (!strcmp(name, "kbtype") ||
   1043           !strcmp(name, "keyboard") ||
   1044               !strcmp(name, "id") ||
   1045                   !strcmp(name, "display") ||
   1046                       !strcmp(name, "clipboard") ||
   1047                           !strcmp(name, "geometry") ||
   1048                               !strcmp(name, "option") ||
   1049                                   !strcmp(name, "options") ||
   1050                                       !strcmp(name, "shadow") ||
   1051                                           !strcmp(name, "shadowmode") ||
   1052                                               !strcmp(name, "streaming") ||
   1053                                                   !strcmp(name, "defer") ||
   1054                                                       !strcmp(name, "tile"))
   1055   {
   1056     argv[1] = value;
   1057     argc = 2;
   1058   }
   1059   else if (!strcmp(name, "R") && !strcmp(value, "1"))
   1060   {
   1061     argc = 1;
   1062   }
...
```

This block is the first in the row, the block I removed will not be processed because "clipboard" matches here.

The check then happens here in ddxProcessArgument:
``` 
    926   if (!strcmp(argv[i], "-clipboard"))
    927   {
    928     if ((!strcmp(argv[i+1], "both")) || (!strcmp(argv[i+1], "1")))
    929     {
    930       nxagentChangeOption(Clipboard, ClipboardBoth);
    931     }
    932     else if (!strcmp(argv[i+1], "client"))
    933     {
    934       nxagentChangeOption(Clipboard, ClipboardClient);
    935     }
    936     else if (!strcmp(argv[i+1], "server"))
    937     {
    938       nxagentChangeOption(Clipboard, ClipboardServer);
    939     }
    940     else if ((!strcmp(argv[i+1], "none")) || (!strcmp(argv[i+1], "0")))
    941     {
    942       nxagentChangeOption(Clipboard, ClipboardNone);
    943     }
    944     else
    945     {
    946       nxagentChangeOption(Clipboard, ClipboardBoth);
    947     }
    948     return 2;
    949   }
```

**Comment by sunweaver** [comment]
@uli42: thanks for explaining this. Then we are good to go.

**Review by sunweaver — APPROVED** [review]

**Pull request merged**